### PR TITLE
Fix docker [WIP]

### LINKED ISSUE #28: Get NWBE docker from OSBv2 building in tests
@Daksh1603 building the NWBE docker image is working locally, but failing in the tests. See https://github.com/OpenSourceBrain/DANDIArchiveShowcase/actions/runs/5124779382/jobs/9217081351

Can you have a look?

### PR DESCRIPTION
**Pull request by Daksh1603**
Closes #28 

The following pull request deals with allowing testing of Dandisets through Docker .

### CODE REVIEW

**Comment by sanjayankur31** [comment]
Sounds good. What error are you seeing? Could you paste the error message and maybe Padraig or I have some idea on what's going on?

**Comment by Daksh1603** [comment]
ERROR: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?

A workaround for this was mounting the host docker.sock to that of the containers.

As for the recent commits, the tests up till running 'generate readme' and running the nwbe container inside the DANDIArchiveShowcase container are working fine.
https://github.com/Daksh1603/DANDIArchiveShowcase/actions/runs/5500841561

It's only the final changes that are pending which are related to testing it on a limited no. of dandisets and possibly tests for other scripts too.

**Comment by Daksh1603** [comment]
Some additional pointers ->
1. Adding lines to generateDocker.sh : This is regarding why I added docker build commands to this shell file in particular . I tried incorporating this inside the dockerfile itself , but for a DIND container to work and essentially run, involves passing the --privileged flag to the container . This allows the container to incorporate all the capabilities of the host in order to build containers inside . However, when building a dockerfile there is no means of incorporating such privileges . After looking on ways if there is any remote support of incorporating this , I found this for [reference](https://github.com/moby/buildkit/blob/master/frontend/dockerfile/docs/reference.md#run---securityinsecure) . This might certainly allow the following with some additional lines in the dockerfile but the disadvantages of the following are : 
+ The user will have to setup and use a different buildkit for docker that allows this to work .
+ Not recommend for CI/CD pipelines .
As an alternative : I Just added these lines to the generateDocker.sh which works well !

2. Mounting the hosts docker.sock : This is a problem that I didn't encounter on my local machine but on github actions . I already mentioned this issue above and the workaround .

3. Persistent data generated for validation is reflected on the host machine as well which is great as any tests run inside the dandiarchive container ( along with the compability_test running inside the nwbe container which itself is inside the dandiarchive container ) generates its result on the host machine DANDIArchiveShowcase folder .

Changes yet to be made : 

Tests are running well , containers are working fine when set up but somehow the compatability_test.py file is not being discovered when testing. This seems like the main point but it would just be a little error that can be fixed within no time . I am trying to troubleshoot this but I am running into this error on my host machine : 

failed to solve with frontend dockerfile.v0: failed to create LLB definition: failed to authorize: rpc error: code = Unknown desc = failed to fetch anonymous token: Get https://auth.docker.io/token?scope=repository%3Alibrary%2Fjava%3Apull&service=registry.docker.io: net/http: TLS handshake timeout

This does happen sometimes when I am working on this, for some reason for like a day in between, when I try to run basic apk update commands, its stuck on cannot fetch try again later . This is able to work after some time, but I am convinced its my ISP acting up sometimes . As soon as I am able to build the container again on my local machine, I would resolve this in no time . As for the other tests, they seem to pass.

File: .github/workflows/ci-conda.yml
Comment (Daksh1603) [inline .github/workflows/ci-conda.yml:1]: --npm-skip will be removed .

File: nwb_table_readme.py [hunk 1]
```diff
@@ -20,6 +20,21 @@
 from dandi import download
 import ast
 from collections import Counter
+import signal
+from contextlib import contextmanager
+
+class TimeoutException(Exception): pass
+
+@contextmanager
```
Comment (pgleeson) [inline nwb_table_readme.py:29]: @Daksh1603 Has this been successfully tested on your side? Checking again here and it still gets stuck around Dandiset 13 with a memory leak, lasting longer than 5 mins.

Screenshot 2023-08-08 at 12 54 47

**Comment by pgleeson** [comment]
@Daksh1603 Any update on the docker functionality since we spoke last week, e.g. blacklisting of problematic dandisets?

**Comment by Daksh1603** [comment]
Hi @pgleeson @sanjayankur31,
Made some changes . The memory issue is no longer being reflected on my machine and hopefully the github actions test can also reflect the same. Please try it out and let me know if the problem still persists .

**Pull request merged**